platform/podjs/bin/genPodjs.js

```diff
@@ -0,0 +1,75 @@
+const MagicString = require("magic-string");
+const fs = require("fs");
+
+function replaceManifestData(code, manifestData) {
+    const magicString = new MagicString(code);
+    const pattern = new RegExp("window.manifestData", "g");
+
+    let hasReplacements = false;
+    let match;
+    let start;
+    let end;
+    let replacement;
+
+    while ((match = pattern.exec(code))) {
+        hasReplacements = true;
+
+        start = match.index;
+        end = start + match[0].length;
+
+        replacement = String(manifestData);
+
+        magicString.overwrite(start, end, replacement);
+    }
+
+    if (!hasReplacements) {
+        return null;
+    }
+
+    return magicString.toString();
+}
+
+function executeReplacement(podJs, manifestJsonPath) {
+    try {
+        const manifestData = fs.readFileSync(manifestJsonPath, "utf8");
+        const podJsCode = fs.readFileSync(podJs, "utf8");
+
+        const replacedCode = replaceManifestData(podJsCode, manifestData);
```

Comment:
Have you considered

```js
const replacedCode = podJsCode.replace( "windows.manifestData", manifestData)
```

Reply:
the issue here is that we don't replace string to string, but string to JSON obj. So it works the same as the previous `imports` in the features

Reply:
Which you then convert to string. In that case we can probably simply convert it to a string here. In other words, how would that be different from the output of current replaceManifestData?

```js
const replacedCode = podJsCode.replace( "windows.manifestData", JSON.stringify(manifestData))
```